src/components/ListItem.js

```diff
@@ -0,0 +1,50 @@
+import { doc, updateDoc } from 'firebase/firestore';
+import { db } from '../lib/firebase';
+import { useEffect, useState } from 'react';
+
+const oneDayInSeconds = 86400;
+
+const style = {
+       listStyleType: 'none',
+       textAlign: 'left'
+}
+const ListItem = ({ itemData }) => {
+    const [checked, setChecked] = useState(itemData.lastPurchased !== null)
+
+    const wasPurchasedWithin24Hours = () => {
+        if (itemData.lastPurchased === null) {
+            return false
+        }
+        return (Math.floor(Date.now() / 1000 ) - itemData.lastPurchased.seconds) <= oneDayInSeconds
+    }
+
+    useEffect(() => {
+        if (itemData.lastPurchased === null) {
+            return false
+        }
+           
+        if ((Math.floor(Date.now() / 1000 ) - itemData.lastPurchased.seconds) >= oneDayInSeconds) {
+            setChecked(false)
+        } else {
+            setChecked(true)
+        }
```

Comment:
I think we can remove the else statement here because if the item is already checked we don't need to set the checked state to true again. I'v tried to remove it and everything still works just fine. let me know if there is something I'm missing .